go.mod

```diff
@@ -4,6 +4,7 @@ go 1.22.1
 
 require (
 	github.com/go-sql-driver/mysql v1.8.1
+	github.com/gorilla/mux v1.8.1
```

Comment:
I updated the router to use gorilla mux instead because it more features than http.NewServeMux. The http ServeMux does not have support for middleware.

Reply:
Sounds good!